cartesapp/cli.py

```diff
@@ -0,0 +1,466 @@
+    imagezero = kwargs.get('imagezero') or "image_0"
+    imagebase = kwargs.get('imagebase') or "image"
+    #
+    flashdrivename = kwargs.get('flashdrivename') or 'data'
+    flashdrivesize = kwargs.get('flashdrivesize') or '128Mb'
+    #
+    blocksize = kwargs.get('blocksize') or 4096
+    return {
+        "sdkversion": sdkversion,
+        "ramsize": ramsize,
+        "datasize": datasize,
+        "basepath": basepath,
+        "imagezero": imagezero,
+        "imagebase": imagebase,
+        "flashdrivename": flashdrivename,
+        "blocksize": blocksize,
+        "flashdrivesize": flashdrivesize
+    }
+
+
+def create_extfs(config):
+    import subprocess
+    su = ["--env",f"USER={os.getlogin()}","--env",f"GROUP={os.getgid()}","--env",f"UID={os.getuid()}","--env",f"GID={os.getgid()}"]
+    args = ["docker","container","run","--rm", 
+            f"--volume={os.getcwd()}/{config['basepath']}:/mnt"]
+    args.extend(su)
+    args.append(f"sunodo/sdk:{config['sdkversion']}")
+    args1 = args.copy()
+    args1.extend(["retar",f"/mnt/{config['imagebase']}.tar"])
+    result = subprocess.run(args1)
+    if result.returncode > 0:
+        raise Exception(f"Error doing retar: {str(result.stderr)}")
+    #
+    #
+    extra_size = parse_size(config['datasize'])//config['blocksize']
+    args2 = args.copy()
+    args2.extend(["genext2fs","--tarball",f"/mnt/{config['imagebase']}.tar",
+                  "--block-size",str(config['blocksize']),"--faketime",
+                  "--readjustment",f"+{extra_size}",f"/mnt/{config['imagebase']}.ext2"])
+    result = subprocess.run(args2)
+    if result.returncode > 0:
+        raise Exception(f"Error generating ext2 fs: {str(result.stderr)}")
+    os.remove(f"{config['basepath']}/{config['imagebase']}.tar")
+    #
+    #
+    flashdrive_bsize = parse_size(config['flashdrivesize'])//config['blocksize']
+    args3 = args.copy()
+    args3.extend(["genext2fs","--faketime","--size-in-blocks", str(flashdrive_bsize),
+                  "--block-size",str(config['blocksize']),f"/mnt/{config['flashdrivename']}.ext2"])
+    result = subprocess.run(args3)
+    if result.returncode > 0:
+        raise Exception(f"Error generating flashdrive ext2 fs: {str(result.stderr)}")
+    
+
+def create_machine_image(config):
+    import subprocess, shutil
+    image0_path = f"{os.getcwd()}/{config['basepath']}/{config['imagezero']}"
+    if os.path.exists(image0_path):
+        shutil.rmtree(image0_path)
+    image_path = f"{os.getcwd()}/{config['basepath']}/{config['imagebase']}"
+    if os.path.exists(image_path):
+        shutil.rmtree(image_path)
+    
+    su = ["--env",f"USER={os.getlogin()}","--env",f"GROUP={os.getgid()}","--env",f"UID={os.getuid()}","--env",f"GID={os.getgid()}"]
+    args = ["docker","container","run","--rm", 
+            f"--volume={os.getcwd()}/{config['basepath']}:/mnt"]
+    args.extend(su)
+    args.append(f"sunodo/sdk:{config['sdkversion']}")
```

Comment:
Can we use the Cartesi Machine Emulator for this? Using the [Cartesi Machine Emulator image](https://hub.docker.com/r/cartesi/machine-emulator) will free us from waiting for Sunodo Image updates, and it probably is a smaller image.

Reply:
We can evaluate, maybe open an issue for that.
The initial idea was to seamless integrate with sunodo, and for now, for the sake of simplicity I think we should use sunodo sdk. We will need the same cartesi machine that the node will run. Since the cm version of the node will be defined by the sunodo sdk, we could use the same image to generate the cm.
And we can always have a custom sunodo sdk version as we did [here](https://github.com/prototyp3-dev/world-arcade-app/blob/f4c74531461513afb6187a8219aa7589cb31e396/Dockerfile#L3).